app/upgrades/v14/upgrades.go

```diff
@@ -16,6 +19,28 @@ func CreateUpgradeHandler(
 	keepers *keepers.AppKeepers,
 ) upgradetypes.UpgradeHandler {
 	return func(ctx sdk.Context, plan upgradetypes.Plan, fromVM module.VersionMap) (module.VersionMap, error) {
+		keepers.SwapRouterKeeper.SetParams(ctx, swaproutertypes.DefaultParams())
+
+		// N.B: pool id in gamm is to be deprecated in the future
+		// Instead,it is moved to swaprouter.
+		migrateNextPoolId(ctx, keepers.GAMMKeeper, keepers.SwapRouterKeeper)
+
+		//  N.B.: this is done to avoid initializing genesis for swaprouter module.
+		// Otherwise, it would overwrite migrations with InitGenesis().
+		// See RunMigrations() for details.
+		fromVM[swaproutertypes.ModuleName] = 0
+
 		return mm.RunMigrations(ctx, configurator, fromVM)
 	}
 }
+
+func migrateNextPoolId(ctx sdk.Context, gammKeeper *gammkeeper.Keeper, swaprouterKeeper *swaprouter.Keeper) {
+	// N.B: pool id in gamm is to be deprecated in the future
+	// Instead,it is moved to swaprouter.
+	nextPoolId := gammKeeper.GetNextPoolId(ctx)
+	swaprouterKeeper.SetNextPoolId(ctx, nextPoolId)
+
+	for poolId := uint64(1); poolId < nextPoolId; poolId++ {
+		swaprouterKeeper.SetModuleRoute(ctx, poolId, swaproutertypes.Balancer)
+	}
```

Comment:
Wait why is it `swaproutertypes.Balancer`, should be `swaproutertypes.Gamm`, many of these pools will be stableswap pools?